File: 이것이 취업을 위한 코딩 테스트다/이진_탐색/부품_찾기/부품_찾기_계수_정렬.java
import java.util.Scanner;

public class 부품_찾기_계수_정렬 {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int[] store = new int[1000001];
    
    for (int i = 0; i < n; i++) {
      int num = sc.nextInt();
      store[num] = 1;
    }

    int m = sc.nextInt();

    for (int i = 0; i < m; i++) {
      int num = sc.nextInt();

      if (store[num] == 1)  {
        System.out.print("yes ");
      } else {
        System.out.print("no ");
      }
    }
  }
}

File: 이것이 취업을 위한 코딩 테스트다/이진_탐색/부품_찾기/부품_찾기_계수_정렬.cpp
#include <iostream>
using namespace std;

int main() {
  int n, m, num;
  int store[1000001] = {0};

  cin >> n;

  for (int i = 0; i < n; i++) {
    cin >> num;
    store[num] = 1;
  }

  cin >> m;
  
  for (int i = 0; i < m; i++) {
    int result = -1;

    cin >> num;
    
    if (store[num] == 1) {
      cout << "yes" << ' ';
    } else {
      cout << "no" << ' ';
    }
  }

  return 0;
}